sdks/go/pkg/beam/core/graph/bind.go

```diff
@@ -249,34 +249,37 @@ func tryBindInbound(candidate typex.FullType, args []funcx.FnParam, isMain bool)
 				}
 			}
 
-		case typex.GBKType:
+		case typex.CoGBKType:
 			if args[0].Kind != funcx.FnValue {
-				return nil, kind, fmt.Errorf("Key of %v cannot bind to %v", t, args[0])
+				return nil, kind, fmt.Errorf("key of %v cannot bind to %v", t, args[0])
```

Comment:
general comment: in these cases, we record why the the thing is not acceptable, but perhaps we could also record what our expectation is. The line number information is not included in these error strings, so it would help making the chain of causality more clear. This has bitten me in the past with coders in particular.